providers/dns/autodns/autodns.go

```diff
@@ -0,0 +1,116 @@
+package autodns
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/go-acme/lego/v3/challenge/dns01"
+	"github.com/go-acme/lego/v3/platform/config/env"
+)
+
+const (
+	envAPIUser            = `AUTODNS_API_USER`
+	envAPIPassword        = `AUTODNS_API_PASSWORD`
+	envAPIEndpoint        = `AUTODNS_ENDPOINT`
+	envAPIEndpointContext = `AUTODNS_CONTEXT`
+	envTTL                = `AUTODNS_TTL`
+
+	defaultEndpoint = `https://api.autodns.com/v1/`
+	demoEndpoint    = `https://api.demo.autodns.com/v1/`
+
+	defaultEndpointContext int = 4
+	defaultTTL             int = 600
+)
+
+type Config struct {
+	Endpoint   *url.URL
+	Username   string `json:"username"`
+	Password   string `json:"password"`
+	Context    int    `json:"-"`
+	TTL        int    `json:"-"`
+	HTTPClient *http.Client
+}
+
+func NewDefaultConfig() *Config {
+	endpoint, _ := url.Parse(defaultEndpoint)
+
+	return &Config{
+		Endpoint:   endpoint,
+		Context:    defaultEndpointContext,
+		HTTPClient: &http.Client{},
+	}
+}
+
+type DNSProvider struct {
+	config *Config
+	//zoneNameservers map[string]string
+	//currentRecords  []*ResourceRecord
+}
+
+func NewDNSProvider() (*DNSProvider, error) {
+	values, err := env.Get(envAPIUser, envAPIPassword)
+	if err != nil {
+		return nil, fmt.Errorf("autodns: %v", err)
+	}
+
+	rawEndpoint := env.GetOrDefaultString(envAPIEndpoint, defaultEndpoint)
+	endpoint, err := url.Parse(rawEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("autodns: %v", err)
+	}
+
+	endpointContext := env.GetOrDefaultInt(envAPIEndpointContext, defaultEndpointContext)
+
+	ttl := env.GetOrDefaultInt(envTTL, defaultTTL)
+
+	config := NewDefaultConfig()
+	config.Username = values[envAPIUser]
+	config.Password = values[envAPIPassword]
+	config.Endpoint = endpoint
+	config.Context = endpointContext
+	config.TTL = ttl
+
+	provider := &DNSProvider{config: config}
+
+	// Because autodns needs the nameservers for each request, we query them all here and put them
```

Comment:
could you follow the same pattern as other provider: `NewDNSProvider`, `NewDNSProviderConfig`

Reply:
Done.